### DISCUSSION TITLE
Is there any way to check if a Bun installation was done through the official script

### DISCUSSION THREAD

**Discussion by SteveLauC** [oven-sh/bun#13605] [Q&A]
Hi, the Bum community!

I am a maintainer of [Topgrade](https://github.com/topgrade-rs/topgrade), a tool that is used to update everything. 

Recently I encountered this [issue](https://github.com/topgrade-rs/topgrade/issues/892), after discussion, I would like to only invoke the update procedure for Bun iff it is installed through the official script. Currently, I determine this by comparing if the Bun binary path is a descendant of the `bun_install_path`, from the install script, `bun_install_path` can be set via environment variable `BUN_INSTALL`, and if not set, defaults to `~/.bun`.

The problem with this approach is that the environment variable `BUN_INSTALL` is not necessary after installation, it is only used during the installation.

I would like to know if there is any reliable way to check if an installation is done through the official script.

**Comment by dotfrag** [comment]
I would like this too, as I maintain different linux machines and it'd be nice to have a way to detect this. It will help for any automated installer/updater scripts and tools, including topgrade. Using this in ansible is another example.
